spectec/spec/6-reduction.watsup

```diff
@@ -172,44 +173,47 @@ rule Step_pure/local.tee:
 
 
 rule Step_read/global.get:
-  z; (GLOBAL.GET x)  ~>  $global(z, x)
+  z; (GLOBAL.GET x)  ~>  $global(z, x).VALUE
 
 rule Step/global.set:
   z; val (GLOBAL.SET x)  ~>  $with_global(z, x, val); epsilon
 
 
 rule Step_read/table.get-trap:
   z; (CONST I32 i) (TABLE.GET x)  ~>  TRAP
-  -- if i >= |$table(z, x)|
+  -- if i >= |$table(z, x).ELEM|
 
 rule Step_read/table.get-val:
-  z; (CONST I32 i) (TABLE.GET x)  ~>  $table(z,x)[i]
-  -- if i < |$table(z, x)|
+  z; (CONST I32 i) (TABLE.GET x)  ~>  $table(z,x).ELEM[i]
+  -- if i < |$table(z, x).ELEM|
 
 rule Step/table.set-trap:
   z; (CONST I32 i) ref (TABLE.GET x)  ~>  z; TRAP
-  -- if i >= |$table(z, x)|
+  -- if i >= |$table(z, x).ELEM|
 
 rule Step/table.set-val:
   z; (CONST I32 i) ref (TABLE.GET x)  ~>  $with_table(z, x, i, ref); epsilon
-  -- if i < |$table(z, x)|
+  -- if i < |$table(z, x).ELEM|
 
 
 rule Step_read/table.size:
   z; (TABLE.SIZE x)  ~>  (CONST I32 n)
-  -- if |$table(z, x)| = n  ;; TODO: inline this
+  -- if |$table(z, x).ELEM| = n  ;; TODO: inline this
 
 
 rule Step/table.grow-succeed:
-  z; ref (CONST I32 n) (TABLE.GROW x)  ~>  $with_tableext(z, x, ref^n); (CONST I32 $(|$table(z, x)|))
+  z; ref (CONST I32 n) (TABLE.GROW x)  ~>  $with_tableinst(z, x, ti'); (CONST I32 $(|$table(z, x).ELEM|))
+  -- if $table(z, x) = ti
+  -- if $grow_table(ti, n, ref) = ti'
+  -- Tabletype_ok: |- ti'.TYPE : OK
```

Comment:
This condition should probably be moved to $grow_table, such that that becomes a self-contained abstraction.